server/banned.go

```diff
@@ -32,46 +34,49 @@ func (pi *packetInfo) checkBlameMessage() error {
 		return nil
 	}
 
+	reason := packet.GetMessage().GetBlame().GetReason()
 	validBlame := false
 
-	for _, reason := range validBlamereasons {
-		if packet.GetMessage().GetBlame().GetReason() == reason {
+	for _, r := range validBlamereasons {
+		if reason == r {
 			validBlame = true
+			break
 		}
 	}
 
-	if validBlame {
+	if !validBlame {
+		return fmt.Errorf("unknown blame reason: %s", reason)
+	} else {
 		blamer := pi.tracker.playerByConnection(pi.conn)
 		if blamer == nil {
+			log.Debugf(logBlame+"Ignoring blame from %s because they have disconnected\n", getIP(pi.conn))
 			return nil
 		}
 		accusedKey := packet.GetMessage().GetBlame().GetAccused().GetKey()
 		accused := blamer.pool.PlayerFromSnapshot(accusedKey)
 		if accused == nil {
-			return errors.New("invalid blame")
-		}
-
-		if accused.pool != blamer.pool {
-			return errors.New("invalid blame")
+			return errors.New("invalid blame - accused not in pool snapshot")
 		}
 
 		// After validating everything, we can skip the actual ban
 		// if the pool already has banned someone.
 		if blamer.pool.firstBan != nil {
+			log.Debugf(logBlame+"Ignoring blame in pool %d because a player is already banned\n", blamer.pool.num)
 			return nil
 		}
 
 		added := accused.addBlame(blamer.verificationKey)
 		if !added {
+			log.Debugf(logBlame+"Duplicate blame\nFrom player: %s\nTo player: %s\n", blamer, accused)
```

Comment:
NIT: What do you think about just using `To:` and `From:`?

Reply:
player doesn't add anything here does it?

e02fe5c